File: src/main/java/com/mpe85/grampa/input/impl/StringInputBuffer.java
package com.mpe85.grampa.input.impl;

import com.mpe85.grampa.input.InputBuffer;

/**
 * An {@link InputBuffer} implementation using a {@link String}.
 * 
 * @author mpe85
 *
 */
public class StringInputBuffer extends CharSequenceInputBuffer {
	
	public StringInputBuffer(final String string) {
		super(string);
		this.string = string;
	}
	
	@Override
	public int getCodePoint(final int index) {
		return string.codePointAt(index);
	}
	
	
	private final String string;
	
}

File: src/main/kotlin/com/mpe85/grampa/input/impl/StringInputBuffer.kt
package com.mpe85.grampa.input.impl

import com.mpe85.grampa.input.InputBuffer

/**
 * An [InputBuffer] implementation using a [String].
 *
 * @author mpe85
 * @property[string] A string backing the input buffer
 */
class StringInputBuffer(private val string: String) : CharSequenceInputBuffer(string) {

  override fun getCodePoint(index: Int) = string.codePointAt(index)

}
